crates/trigger/src/stdio.rs

```diff
@@ -56,13 +56,23 @@ impl StdioLoggingTriggerHooks {
         &self,
         component_id: &str,
         log_suffix: &str,
-        log_dir: &Path,
+        log_dir: Option<&Path>,
     ) -> Result<ComponentStdioWriter> {
         let sanitized_component_id = sanitize_filename::sanitize(component_id);
-        let log_path = log_dir.join(format!("{sanitized_component_id}_{log_suffix}.txt"));
+        let mut log_path: Option<PathBuf> = None;
+        if let Some(log_dir) = log_dir {
```

Comment:
Is this fancy talk for `Option::map`?

Reply:
imo combinators shouldn't be mixed with mutation. :shrug:

Reply:
Actually I'm skeptical of combinators any time you throw away the result.

Reply:
I meant it seemed like this whole stanza seemed equivalent to `let log_path = log_dir.map(...)` (sorry I should probably have put the comment on line 62) but if there is mutation of something with larger scope going on then ignore me I completely agree.

Reply:
Oh maybe you are correct. I was confused by `log_path` vs `log_dir` above.